src/mame/irem/olibochu.cpp

```diff
@@ -235,61 +247,50 @@ void olibochu_state::video_start()
 	m_bg_tilemap->set_transparent_pen(0);
 }
 
-void olibochu_state::draw_sprites(bitmap_ind16 &bitmap, const rectangle &cliprect)
+void olibochu_state::draw_sprites(bitmap_ind16 &bitmap, rectangle const &cliprect)
 {
-	// 8x8 sprites
-	for (int offs = m_spriteram[0].bytes() - 4; offs >= 0; offs -= 4)
+	auto draw = [this](bitmap_ind16 &bitmap, rectangle const &cliprect, int const slot, bool const large)
 	{
-		const u8 *src = &m_spriteram[0][offs];
-		int code = src[0];
-		int attr = src[1];
-		int color = attr & 0x3f;
-		int flipx = attr & 0x40;
-		int flipy = attr & 0x80;
-		int sy = src[2];
-		int sx = src[3];
-
+		int const bank = large ? 1 : 0;
+		int const offs = slot * 4;
+		u8 const code  = m_spriteram[bank][offs + 0];
+		u8 const attr  = m_spriteram[bank][offs + 1];
+		u8 const color = BIT(attr, 0, 6);
```

Comment:
IMO for simple masks like that, you don't need the 3-param BIT macro, just use x & 0x3f.